### PR TITLE
Update test ES 6.8 to use es-oss image

### PR DESCRIPTION
**Pull request by AndreKurait**
### Description

I encountered startup issues running the ES 6.8 image on apple silicon due to x-pack-ml.

Fixed encountered metadata issue when only a single template exists in the cluster. 

* Category: Bug Fix
* Why these changes are required? We shouldn't depend on 6.8 x-pack in unit tests
* What is the old behavior before changes and new behavior after changes? Encountered failure `org.elasticsearch.bootstrap.StartupException: org.elasticsearch.bootstrap.BootstrapException: java.io.IOException: Cannot run program "/usr/share/elasticsearch/modules/x-pack-ml/platform/linux-x86_64/bin/controller": error=0, Failed to exec spawn helper: pid: 111, exit value: 1 at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:163)` exiting container in test

### Issues Resolved
[MIGRATIONS-2202](https://opensearch.atlassian.net/browse/MIGRATIONS-2202)

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
Ran locally on M3 Pro

### Check List
- [x] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: RFS/src/testFixtures/java/org/opensearch/migrations/bulkload/framework/SearchClusterContainer.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class SearchClusterContainer extends GenericContainer<SearchClusterContai
         Version.fromString("ES 7.17.22")
     );
     public static final ContainerVersion ES_V6_8_23 = new ElasticsearchVersion(
-        "docker.elastic.co/elasticsearch/elasticsearch:6.8.23",
+        "docker.elastic.co/elasticsearch/elasticsearch-oss:6.8.23",
```
Comment (peternied) [inline RFS/src/testFixtures/java/org/opensearch/migrations/bulkload/framework/SearchClusterContainer.java:33]: Could you move all the ES versions to OSS and create a jira for adding tests for non-OSS ES versions in our OE epic?

**Review by peternied — APPROVED** [review]

File: RFS/src/testFixtures/java/org/opensearch/migrations/bulkload/framework/SearchClusterContainer.java (hunk 1 above)
Comment (AndreKurait) [reply RFS/src/testFixtures/java/org/opensearch/migrations/bulkload/framework/SearchClusterContainer.java:33]: This does that as possible, there is no oss version post 7.10
https://opensearch.atlassian.net/browse/MIGRATIONS-2203

File: RFS/src/main/java/org/opensearch/migrations/bulkload/version_universal/RemoteReaderClient.java [hunk 2]
```diff
@@ -121,8 +121,8 @@ Mono<ObjectNode> getJsonForTemplateApis(HttpResponse resp) {
         try {
             var tree = (ObjectNode) objectMapper.readTree(resp.body);
 
-            if (tree.size() == 1) {
-                return Mono.just(handleSingleItemTree(tree));
+            if (tree.size() == 1 && tree.fields().next().getValue().isArray()) {
```
Comment (peternied) [inline RFS/src/main/java/org/opensearch/migrations/bulkload/version_universal/RemoteReaderClient.java:124]: Whoa, what broke here?  From these changes the shape of the response is different between the OSS and general product line?

Comment (peternied) [reply RFS/src/main/java/org/opensearch/migrations/bulkload/version_universal/RemoteReaderClient.java:124]: This is a good candidate to adding a unit test, LMK if you'd prefer I bootstrap these tests.

Comment (AndreKurait) [reply RFS/src/main/java/org/opensearch/migrations/bulkload/version_universal/RemoteReaderClient.java:124]: The shape isn't different between oss and non-oss, it's just that we previously weren't testing the case of 6.8 with only one item

Getting:
```
{"simple_index_template":{"order":0,"index_patterns":["blog*"],"settings":{"index":{"number_of_shards":"1"}},"mappings":{"_doc":{"_source":{"enabled":true},"properties":{"created_at":{"format":"EEE MMM dd HH:mm:ss Z yyyy","type":"date"},"host_name":{"type":"keyword"}}}},"aliases":{"alias1":{}}}}
```

Whereas 7.10 gives
```
{"component_templates":[{"name":"simple_component_template","component_template":{"template":{"settings":{"index":{"number_of_shards":"1","number_of_replicas":"1"}},"mappings":{"properties":{"author":{"type":"text"}}},"aliases":{"alias1":{}}},"version":1}}]}
```

Before, with non-oss we were hitting the case where size was > 1

Comment (AndreKurait) [reply RFS/src/main/java/org/opensearch/migrations/bulkload/version_universal/RemoteReaderClient.java:124]: Will merge and follow up if any unit tests can be added

**Pull request merged**